keras_cv/training/contrastive/contrastive_trainer.py

```diff
@@ -0,0 +1,260 @@
+        projector,
+        include_probe,
+        classes=None,
+    ):
+        super().__init__()
+
+        if encoder.output.shape.rank != 2:
+            raise ValueError("Encoder must have a flattened output")
+
+        if include_probe:
+            if not classes:
+                raise ValueError(
+                    "`classes` must be specified when `include_probe` is `True`."
+                )
+
+        if type(augmenter) is tuple and len(augmenter) != 2:
+            raise ValueError(
+                "`augmenter` must be either a single augmenter or a tuple of exactly 2 augmenters."
+            )
+
+        if type(projector) is tuple and len(projector) != 2:
+            raise ValueError(
+                "`augmenter` must be either a single augmenter or a tuple of exactly 2 augmenters."
+            )
+
+        self.augmenters = (
+            augmenter if type(augmenter) is tuple else (augmenter, augmenter)
+        )
+        self.encoder = encoder
+        self.projectors = (
+            projector if type(projector) is tuple else (projector, projector)
+        )
+
+        self.loss_metric = keras.metrics.Mean(name="loss")
+
+        self.include_probe = include_probe
+
+        if self.include_probe:
+            self.probing_top = layers.Dense(classes, name="linear_probe")
+            self.probe_loss_metric = keras.metrics.Mean(name="probe_loss")
+            self.probe_metrics = []
+
+    def compile(
+        self, probe_optimizer=None, probe_loss=None, probe_metrics=None, **kwargs
+    ):
+        super().compile(**kwargs)
+
+        if self.include_probe and not probe_optimizer:
+            raise ValueError(
+                "`probe_optimizer` must be specified when `include_probe` is `True`."
+            )
+
+        if self.include_probe and not probe_loss:
+            raise ValueError(
+                "`probe_loss` must be specified when `include_probe` is `True`."
+            )
+
+        if self.include_probe:
+            self.probe_loss = probe_loss
+            self.probe_optimizer = probe_optimizer
+            self.probe_metrics = probe_metrics or []
+
+    @property
+    def metrics(self):
+        metrics = [
+            self.loss_metric,
+        ]
+        if self.include_probe:
+            metrics += [self.probe_loss_metric]
+            metrics += self.probe_metrics
+        return super().metrics + metrics
+
+    def fit(
+        self,
+        x=None,
+        y=None,
+        sample_weight=None,
+        batch_size=None,
+        **kwargs,
+    ):
+        dataset = _convert_inputs_to_tf_dataset(
+            x=x, y=y, sample_weight=sample_weight, batch_size=batch_size
+        )
+
+        dataset = dataset.map(self.run_augmenters, num_parallel_calls=tf.data.AUTOTUNE)
+
+        return super().fit(x=dataset, **kwargs)
+
+    def run_augmenters(self, x, y=None):
+        if y is not None:
+            return (
+                x,
+                self.augmenters[0](x, training=True),
```

Comment:
Let's use the KerasCV construct of:

```
inputs = {"images": x}
if y:
    inputs["labels"] = y

i1 = self.augmenters[0](x, training=True)
i2 = self.augmenters[0](x, training=True)
```

then we can return from the output dictionaries

Reply:
the reason for this is y can change too in some cases based on the augmentation technique, also it can cut out the the conditional I believe.

Reply:
Sounds good -- Done.
I've keyed the augmented image sets as "augmented_images_{0/1}"